parser/parser/include/ruby_parser/driver.hh

```diff
@@ -76,6 +76,22 @@ struct node_with_token {
     ForeignPtr nod = nullptr;
 };
 
+struct node_with_context {
```

Comment:
Whitequark now returns multiple values from its parsing rules, node itself and a context object. We can represent this using 2 new `struct`s: `node_with_context` and `token_with_context`.

https://github.com/whitequark/parser/commit/09d681e534885f1aa22f0099089841ae9d86f847#diff-9790c9a058dad53182a4e531a3b0b724077fba6ee069ceb5845ae2dfc2398814R266